src/pytest_html/scripts/main.js

```diff
@@ -78,31 +63,26 @@ const renderDerived = (tests, collectedItems) => {
 
     const currentFilter = getFilter()
     possibleOutcomes.forEach(({ outcome, label }) => {
-        const count = renderSet.filter((test) => {
-            const wasXpassed = outcome === 'xpassed' && ['passed', 'failed'].includes(test.outcome)
-            const wasXfailed = outcome === 'xfailed' && test.outcome === 'skipped'
-            if (typeof test.wasxfail !== 'undefined') {
-                return wasXpassed || wasXfailed
-            } else {
-                return test.outcome === outcome
-            }
-        }).length
+        const count = renderSet.filter((test) => test.outcome.toLowerCase() === outcome).length
         const input = document.querySelector(`input[data-test-result="${outcome}"]`)
         document.querySelector(`.${outcome}`).innerText = `${count} ${label}`
 
         input.disabled = !count
         input.checked = !currentFilter.includes(outcome)
     })
 
-    if (collectedItems === renderSet.length) {
+    const numberOfTests = renderSet.filter(({outcome}) =>
+      ["Passed", "Failed", "XPassed", "XFailed"].includes(outcome)
+    ).length
+    if (runningState === "Finished") {
```

Comment:
how about creating a isDone fn in datamanager?
That way we wont have a need for comparing strings here.